source/kraken/fill_disruption_from_chaos.cpp

```diff
@@ -46,6 +48,28 @@ namespace nt = navitia::type;
 namespace bt = boost::posix_time;
 namespace bg = boost::gregorian;
 
+namespace {
+
+template<typename T>
+std::vector<std::unique_ptr<T>>& get_vj_list_of_jp_helper(nt::JourneyPattern* jp);
+
+template<>
+auto get_vj_list_of_jp_helper<nt::DiscreteVehicleJourney>(nt::JourneyPattern* jp)
+-> decltype((jp->discrete_vehicle_journey_list))
+{ return jp->discrete_vehicle_journey_list; }
+
+template<>
+auto get_vj_list_of_jp_helper<nt::FrequencyVehicleJourney>(nt::JourneyPattern* jp)
+-> decltype((jp->frequency_vehicle_journey_list))
+{ return jp->frequency_vehicle_journey_list; }
+
+template<typename Container, typename Value>
```

Comment:
Ok.
So the real question here is to use sets of vj in meta_vj and a set of jp in Impact.